src/qibo_cloud_backends/aws_client.py

```diff
@@ -0,0 +1,232 @@
+        else:
+            return "Unknown"
+    
+    def remove_qelib1_inc(self, qasm_string):
+        """To remove the 'includes qe1lib.inc' from the OpenQASM string.
+
+        Args: 
+            qasm_code (OpenQASM circuit, str): circuit given in the OpenQASM format.
+
+        Returns:
+            qasm_code (OpenQASM circuit, str): circuit given in the OpenQASM format.
+        """
+        
+        # Remove the "include "qelib1.inc";\n" line
+        modified_code = re.sub(r'include\s+"qelib1.inc";\n', '', qasm_string)
+        return modified_code
+    
+    def qasm_convert_gates(self, qasm_code):
+        """To replace the notation for certain gates in OpenQASM
+
+        Args: 
+            qasm_code (OpenQASM circuit, str): circuit given in the OpenQASM format.
+
+        Returns:
+            qasm_code (OpenQASM circuit, str): circuit given in the OpenQASM format.
+        """
+        
+        lines = qasm_code.split('\n')
+        modified_code = ""
+        for line in lines:
+            for key in _QASM_BRAKET_GATES:
+                if key in line:
+                    line = line.replace(key, _QASM_BRAKET_GATES[key])
+                    break
+            modified_code += line + '\n'
+        return modified_code
+    
+    def transpile_qasm_circuit(self, circuit_qasm, native_gates, custom_coupling_map, optimization_level=1):
+        """Transpiles a circuit given in OpenQASM format using Qiskit's transpiler.
+            Returns a circuit given in OpenQASM format.
+
+        Args:
+            circuit_qasm (OpenQASM circuit, str): circuit given in the OpenQASM format.
+            native_gates (list, str): A list of strings representing the native gates of the QPU.
+                e.g. native_gates = ['ecr', 'i', 'rz', 'sx', 'x']
+            custom_coupling_map (list, list): A list containing lists representing the connectivity of the qubits.
+                e.g. custom_coupling_map. E.g. [[0, 1], [1, 2], [2, 3]]
+            optimization_level (int): Optimization level for Qiskit's transpiler. Range is from 0 to 3. Defaults to 1.
+
+        Returns:
+            transpiled_circuit_qasm (OpenQASM circuit, str): Transpiled circuit in the OpenQASM format.
+        """
+
+        self.native_gates = native_gates
+        self.custom_coupling_map = custom_coupling_map
+
+        if optimization_level < 0 or optimization_level > 3:
+            raise_error(ValueError, "Optimization_level is between 0 to 3.")
+        if custom_coupling_map is None:
+            raise_error(ValueError, "Expected custom_coupling_map. E.g. custom_coupling_map = [[0, 1], [1, 2], [2, 3]]")
+        if native_gates is None:
+            raise_error(ValueError, "Expected native gates for transpilation. E.g. native_gates = ['ecr', 'i', 'rz', 'sx', 'x']")
+
+        # Convert to Qiskit circuit for transpilation
+        circuit_qiskit = QuantumCircuit.from_qasm_str(circuit_qasm)
+        transpiled_circuit = transpile(circuit_qiskit,
```

Comment:
If you're already using Qiskit's `transpile` function, why not use the Qiskit-Braket Provider's [`to_braket`](https://github.com/qiskit-community/qiskit-braket-provider/blob/main/qiskit_braket_provider/providers/adapter.py#L397) function instead? That way, you won't need the `remove_qelib1_inc` or `qasm_convert_gates` functions, and you don't need to directly call `from_ir` either.

That said, why not use Qibo's own transpile function?